tests/test_parser.py

```diff
@@ -2085,3 +2085,76 @@ def test_string_attribute_reference_5():
     assert instance1.name == "x"
     assert str(instance1.locatable_name) == "x"
     assert instance1.locatable_name.location == Range("test", 5, 1, 5, 2)
+
+
+@pytest.mark.parametrize_any(
+    "snippet",
+    [
+        "mymod.MyEntity()",
+        "mymod.submod.MyEntity()",
+        "mymod.submod.MyEntity(x=1)",
+        "mymod.submod.MyEntity(**dct)",
+        "entity Child extends mymod.MyEntity: end",
+        "SomeEntity.my [1] -- mymod.MyEntity",
+    ],
+)
+def test_entity_ref_err_dot(snippet: str) -> None:
+    """
+    Verify that an attempt to access an entity in a qualified manner with '.' instead of '::' results in an appropriate
+    exception.
+
+    :param snippet: Snippet that is expected to produce this error for an entity named `MyEntity`.
+    """
+    with pytest.raises(ParserException, match="`MyEntity` looks like an entity but was accessed with '.'"):
+        parse_code(snippet)
+
+
+def test_entity_ref_err_dot_full_msg() -> None:
+    with pytest.raises(
+        ParserException,
+        match=re.escape(
+            "Syntax error: `MyEntity` looks like an entity but was accessed with '.' (`mymod.submod.MyEntity`)."
+            " To access an entity in a namespace, use '::' instead: `mymod::submod::MyEntity` (test:2:5)"
```

Comment:
I think I would like the following structure more

```suggestion
            "Syntax error: `mymod.submod.MyEntity` looks like an entity but was accessed with '.'  instead of '::'."
            " Use `mymod::submod::MyEntity` instead.(test:2:5)"
```

Reply:
(as a comment to the side, we may want to distinguish the suggestion at the class level, because that would allow us to implement a 'quick-fix' in vscode based on the info we already have (range+suggestion) )